tests/lsdb/loaders/dataframe/test_from_dataframe.py

```diff
@@ -0,0 +1,144 @@
+):
+    """Tests that specifying simultaneously threshold and partition_size is invalid"""
+    kwargs = get_catalog_kwargs(small_sky_order1_catalog, partition_size=10, threshold=10_000)
+    with pytest.raises(ValueError, match="Specify only one: threshold or partition_size"):
+        lsdb.from_dataframe(small_sky_order1_df, **kwargs)
+
+
+def test_partitions_on_map_equal_partitions_in_df(small_sky_order1_df, small_sky_order1_catalog):
+    """Tests that partitions on the partition map exist in the Dask Dataframe"""
+    kwargs = get_catalog_kwargs(small_sky_order1_catalog)
+    catalog = lsdb.from_dataframe(small_sky_order1_df, **kwargs)
+    for _, partition_index in catalog._ddf_pixel_map.items():
+        partition_df = catalog._ddf.partitions[partition_index].compute()
+        assert isinstance(partition_df, pd.DataFrame)
+
+
+def test_partitions_in_partition_info_equal_partitions_on_map(small_sky_order1_df, small_sky_order1_catalog):
+    """Tests that partitions in the partition info match those on the partition map"""
+    kwargs = get_catalog_kwargs(small_sky_order1_catalog)
+    catalog = lsdb.from_dataframe(small_sky_order1_df, **kwargs)
+    for _, row in catalog.hc_structure.get_pixels().iterrows():
+        hp_pixel = HealpixPixel(row["Norder"], row["Npix"])
+        partition_from_df = catalog.get_partition(hp_pixel.order, hp_pixel.pixel)
+        partition_index = catalog._ddf_pixel_map[hp_pixel]
+        partition_from_map = catalog._ddf.partitions[partition_index]
+        pd.testing.assert_frame_equal(partition_from_df.compute(), partition_from_map.compute())
+
+
+def test_partitions_on_map_match_pixel_tree(small_sky_order1_df, small_sky_order1_catalog):
+    """Tests that HEALPix pixels on the partition map exist in pixel tree"""
+    kwargs = get_catalog_kwargs(small_sky_order1_catalog)
+    catalog = lsdb.from_dataframe(small_sky_order1_df, **kwargs)
+    for hp_pixel, _ in catalog._ddf_pixel_map.items():
+        if hp_pixel in catalog.hc_structure.pixel_tree:
+            assert catalog.hc_structure.pixel_tree[hp_pixel].node_type == PixelNodeType.LEAF
+
+
+def test_from_dataframe_with_non_default_ra_dec_columns(small_sky_order1_df, small_sky_order1_catalog):
+    """Tests the creation of a catalog using non-default ra and dec columns"""
+    kwargs = get_catalog_kwargs(small_sky_order1_catalog, ra_column="my_ra", dec_column="my_dec")
+    # If the columns for ra and dec do not exist
+    with pytest.raises(KeyError):
+        lsdb.from_dataframe(small_sky_order1_df, **kwargs)
+    # If they were indeed named differently
+    small_sky_order1_df.rename(columns={"ra": "my_ra", "dec": "my_dec"}, inplace=True)
+    lsdb.from_dataframe(small_sky_order1_df, **kwargs)
+
+
+def test_partitions_obey_partition_size(small_sky_order1_df, small_sky_order1_catalog):
+    """Tests that partitions are limited by the specified size"""
+    partition_size_bytes = 1 << 10
+    partition_size_megabytes = partition_size_bytes / (1 << 20)
```

Comment:
Pretty sure this is equivalent to `partition_size_bytes >> 20`

Reply:
We really need to perform the division because using right shift we would not get the partition size in megabytes as a float. In this case we are trying to partition by 1KB so that would happen.